src/v3x/register-instructions.cpp

```diff
@@ -13,39 +13,40 @@ void register_instructions(analyzer::Analyzer *analyzer) {
     assert(analyzer);
 
     const auto& instruction_set = InstructionSet::get_instance();
-    const auto& gate_map = instruction_set.get_gate_map();
+    const auto& named_gate_map = instruction_set.get_named_gate_map();
     const auto& non_gate_map = instruction_set.get_non_gate_map();
 
-    // Gates
-    for (const auto &[gate_name, pair_params_types_operands_types] : gate_map) {
+    // Named gates
+    for (const auto &[gate_name, pair_params_types_operands_types] : named_gate_map) {
         const auto &operand_types = pair_params_types_operands_types.second;
         analyzer->register_instruction(gate_name, operand_types);
     };
 
-    // Single-qubit unitary gates, e.g., inv(X), pow(X, 2), or pow(inv(X), 2)
-    for (const auto &[gate_name, pair_params_types_operands_types] : gate_map) {
-        if (instruction_set.is_single_qubit_gate(gate_name)) {
+    // Single-qubit gates, e.g., inv.X, pow(2).X, or pow(2).inv.X
+    for (const auto &[named_gate_name, pair_params_types_operands_types] : named_gate_map) {
+        if (instruction_set.is_single_qubit_named_gate(named_gate_name)) {
             const auto &operand_types = pair_params_types_operands_types.second;
-            const auto &single_qubit_unitary_gate_name = fmt::format("{}_{}",
-                instruction_set.single_qubit_unitary_gate_composition_prefix, gate_name);
-            analyzer->register_instruction(single_qubit_unitary_gate_name, operand_types);
+            const auto &single_qubit_gate_name = fmt::format("{}_{}",
+                instruction_set.single_qubit_gate_composition_prefix, named_gate_name);
+            analyzer->register_instruction(single_qubit_gate_name, operand_types);
         }
     };
 
-    // Two-qubit unitary gates, e.g., ctrl(X), ctrl(inv(X)), ctrl(pow(X, 2)), or ctrl(pow(inv(X), 2))
-    for (const auto &[gate_name, pair_params_types_operands_types] : gate_map) {
-        if (instruction_set.is_single_qubit_gate(gate_name)) {
-            // For a single qubit gate such as Ry, the gate map will hold two entries, namely "Ry Qf" and "Ry Vf"
-            // For each of those entries, e.g., "Ry Qf", the code below registers 2 two-qubit gates "2q_Ry",
-            // each acting on a pair of qubit operands, one on "QQf" and the other on "VQf"
+    // Two-qubit gates, e.g., ctrl(X), ctrl(inv(X)), ctrl(pow(2).X), or ctrl(pow(2).inv(X))
```

Comment:
Is this not cQASM on purpose? Should it be?: 
```suggestion
    // Two-qubit gates, e.g., ctrl.X, ctrl.inv(X), ctrl.pow(2).X, or ctrl.pow(2).inv(X)
```

Reply:
Oh, no, I still didn't write it properly. It should be cQASM on purpose. But that means `ctrl.inv.X`, `inv.X`... only parentheses for`pow`'s parameter.